src/utils/utils.ts

```diff
@@ -2,11 +2,46 @@ import { CustomEventId } from './constants'
 import { CustomEvents } from '../extension/injected-script/addEventListener'
 
 import { sleep as _sleep, timeout as _timeout } from '@holoflows/kit/es/util/sleep'
-import { isNull } from 'lodash-es'
+import { defaults, isNull } from 'lodash-es'
 
 export const sleep = _sleep
 export const timeout = _timeout
 
+/**
+ * while (expression) {
+ *     do();
+ *     sleep(opt.interval)
+ * }
+ */
+const finiteWhileDoSleepDefault = { interval: 1000, timeout: 30000 }
+interface finiteWhileDoSleep {
+    interval?: number
+    timeout?: number
+}
+export const finiteWhileDoSleep = (
```

Comment:
I didn't find where do you use this util. maybe remove it?

Reply:
removed. it is designed to be used in auto paste.